source/server/config/network/http_connection_manager.h

```diff
@@ -48,10 +49,19 @@ class HttpFilterConfigFactory {
 public:
   virtual ~HttpFilterConfigFactory() {}
 
-  virtual HttpFilterFactoryCb tryCreateFilterFactory(HttpFilterType type, const std::string& name,
-                                                     const Json::Object& config,
-                                                     const std::string& stat_prefix,
-                                                     Server::Instance& server) PURE;
+  /**
+  * Create a particular http filter factory implementation.  If the creation fails, an error will be
+  * thrown.  The returned callback should always be valid.
+  */
+  virtual HttpFilterFactoryCb createFilterFactory(HttpFilterType type, const Json::Object& config,
+                                                  const std::string& stat_prefix,
+                                                  Server::Instance& server) PURE;
+
+  /**
+  * Provides the identifying name for a particular implementation of an http filter produced by the
+  * factory.
+  */
+  virtual std::string name() PURE;
```

Comment:
yeah, we could do static const string in a class.
Thinking about this more, as it's called once there is no point of having ref returned, agreed.